Add bytes::Bytes implementation

vorot93 opened the pull request:
Fixes #259

bkchr commented on src/bytes.rs:
```diff
@@ -0,0 +1,13 @@
+use crate::{Decode, Encode, EncodeLike, Error, Input};
```
Missing license header. (just copy from one of the other files).

And could you please also add at least some basic test?

vorot93 commented:
@bkchr I reworked the PR to make use of WrapperTypeEncode and EncodeLike, just the way SCALE is implemented for String and memory containers. Also added tests.

vorot93 commented:
Any chance this could be merged soon? @bkchr

bkchr commented on Cargo.toml:
```diff
@@ -14,6 +14,7 @@ arrayvec = { version = "0.7", default-features = false }
 serde = { version = "1.0.102", optional = true }
 parity-scale-codec-derive = { path = "derive", version = "2.3.1", default-features = false, optional = true }
 bitvec = { version = "0.20.1", default-features = false, features = ["alloc"], optional = true }
+bytes-crate = { package = "bytes", version = "1", default-features = false, optional = true }
```
```suggestion
bytes = { version = "1", default-features = false, optional = true }
```

bkchr commented on Cargo.toml:
```diff
@@ -39,6 +40,7 @@ default = ["std"]
 derive = ["parity-scale-codec-derive"]
 std = ["serde", "bitvec/std", "byte-slice-cast/std", "chain-error"]
 bit-vec = ["bitvec"]
+bytes = ["bytes-crate"]
```
```suggestion
```

bkchr commented on src/codec.rs:
```diff
@@ -1429,6 +1446,19 @@ mod tests {
+	#[cfg(feature = "bytes")]
+	#[test]
+	fn bytes_works_as_expected() {
+		let input = bytes_crate::Bytes::from_static(b"hello");
```
```suggestion
		let input = bytes::Bytes::from_static(b"hello");
```

bkchr commented on src/codec.rs:
```diff
@@ -1429,6 +1446,19 @@ mod tests {
+		assert_eq!(encoded, encoded_vec);
+
+		assert_eq!(
+			bytes_crate::Bytes::decode(&mut (&encoded as &[u8])).unwrap(),
+			Vec::<u8>::decode(&mut (&encoded as &[u8])).unwrap()
```
```suggestion
			&*b"hello"[..],
			bytes::Bytes::decode(&mut &encoded[..]).unwrap(),
```

bkchr commented on src/codec.rs:
```diff
@@ -371,6 +371,23 @@ mod feature_full_wrapper_type_encode {
+#[cfg(feature = "bytes")]
+mod feature_wrapper_bytes {
+	use super::*;
+	use bytes_crate::Bytes;
```
```suggestion
	use bytes::Bytes;
```

bkchr reviewed (approved):
Sorry for the delay

bkchr commented on src/codec.rs:
```diff
@@ -1429,6 +1446,19 @@ mod tests {
+		assert_eq!(encoded, encoded_vec);
+
+		assert_eq!(
+			&*b"hello"[..],
```
```suggestion
			&b"hello"[..],
```

bkchr commented:
Ty for fixing the ui test!

shawntabrizi commented:
@vorot93 if you post your polkadot or kusama address, I will open a treasury tip for you

vorot93 commented:
@shawntabrizi Sure

`15rc4P2NQ3zuzk2r3RYyYtYmCV7aeMGbRdEHxrT2UNdJQpfr` on Polkadot

Thanks!

shawntabrizi commented:
Done: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/treasury/tips
